cmd/krew/cmd/index.go

```diff
@@ -62,14 +67,67 @@ var indexAddCmd = &cobra.Command{
 	Example: "kubectl krew index add default " + constants.IndexURI,
 	Args:    cobra.ExactArgs(2),
 	RunE: func(_ *cobra.Command, args []string) error {
-		return indexoperations.AddIndex(paths.IndexBase(), args[0], args[1])
+		return indexoperations.AddIndex(paths, args[0], args[1])
 	},
 }
 
+var indexDeleteCmd = &cobra.Command{
+	Use:   "remove",
+	Short: "Remove a configured index",
+	Long: `Remove a configured index repository which is used to discover and
+install plugins from.
+
+If there are plugins installed from the specified index, the index cannot be
+removed (unless --force) is used. Removing indexes while there are plugins
+installed from them is not supported behavior.`,
+	Args: cobra.ExactArgs(1),
+	RunE: indexDelete,
+}
+
+func indexDelete(_ *cobra.Command, args []string) error {
+	name := args[0]
+
+	pl, err := installation.InstalledPluginsFromIndex(environment.MustGetKrewPaths().InstallReceiptsPath(), name)
+	if err != nil {
+		return errors.Wrap(err, "failed querying plugins installed from the index")
+	}
+	klog.V(4).Infof("Found %d plugins from index", len(pl))
+
+	if len(pl) > 0 && !*forceIndexDelete {
+		var names []string
+		for _, v := range pl {
+			names = append(names, v.Name)
+		}
+
+		warning := color.New(color.FgRed, color.Bold).Sprint("WARNING:")
+		klog.Warningf("%s Plugins [%s] are still installed from index %q.", warning, strings.Join(names, ", "), name)
+		klog.Warning("Uninstall them first before removing this index (or, use --force, which may result in unsupported behavior).")
+		return errors.Errorf("refusing to remove due to installed plugins from this index")
+	}
+
+	err = indexoperations.DeleteIndex(environment.MustGetKrewPaths(), name)
+	if os.IsNotExist(err) {
+		if *forceIndexDelete {
+			klog.V(4).Infof("Index not found, but --force is used, so not returning an error")
+			return nil // success if --force specified and index does not exist.
+		}
+		return errors.Errorf("index %q does not exist", name)
+	}
+	return errors.Wrap(err, "error while removing the plugin index")
+}
+
+var (
+	forceIndexDelete *bool
+)
+
 func init() {
+	forceIndexDelete = indexDeleteCmd.Flags().Bool("force", false, "Remove index even if it has plugins installed (may yield in unsupported behavior)")
+
+	indexCmd.AddCommand(indexAddCmd)
+	indexCmd.AddCommand(indexListCmd)
+	indexCmd.AddCommand(indexDeleteCmd)
```

Comment:
What's the reason for moving these out of the env var gate? If `indexCmd` isn't being made available can't we just skip these?

Reply:
Practically makes no difference. So yeah maybe I shouldn't have changed this but it was prior to your patch so I kept it, it's harmless.